### ISSUE TITLE
Add possibility to rename classes and/or methods to prevent errors "Class 'xxx' clashes with package of same name"

### ISSUE THREAD

**Issue by lospejos** [leibnitz27/cfr#259]
# Problem solved by this feature
Add possibility to rename classes and/or methods to prevent errors "Class 'xxx' clashes with package of same name".
After decompilation of obfuscated bytecode (f.e. using yGuard), JAR file may contain files and folders (packages) having the same name.
F.e. JAR file may contain package `a.b.c` and at the same time can contain class `a.b.c` (class named `c` in package `a.b`).
I've tried to decompile such JAR using  `--rename true --antiobf true --recovertypeclash true`, but this not helped: source code after decompilation contains the same package name and class name.

# Feature description

It could be a separate parameter/command line switch, or some improvement in renaming strategy when `--rename` parameter is set to `true`. When this parameter/switch is `true`, CFR should analyze packages and class names, and rename classes correctly to prevent above-mentioned error.

# Considerations

It also could be that I just missed some existing parameter, which resolves my requirement. In that case I'm very sorry and kindly ask you to point me on that existing parameter.

# Alternatives

**Comment by leibnitz27** [comment]
Agree,  there's definitely scope for improving collision detection, will see what I can do.

One parameter (You probably don't want to create this by hand as it's long and tedious (unless you only have a few collisions), but I mention it because it's not really documented) that might help is the ability to use an obfuscation mapping file:

```
C:\code\cfr\target\classes>type c:\temp\unmap.txt
my.HelloMumException -> java.lang.IllegalArgumentException:
clever.real.package.FakeThing -> java.lang.Object:
    void wibble() -> finalize
    void tapToes(long,int) -> wait
```

java org.benf.cfr.reader.Main java.lang.Object --obfuscationpath c:\temp\unmap.txt

```
......
    public final native void wait(long var1) throws InterruptedException;

    public final void tapToes(long l, int n) throws InterruptedException {
        if (l < 0L) {
            throw new HelloMumException("timeout value is negative");
        }
        if (n < 0 || n > 999999) {
            throw new HelloMumException("nanosecond timeout value out of range");
        }
        if (n > 0) {
            ++l;
        }
        this.wait(l);
    }

    @Deprecated(since="9")
    protected void wibble() throws Throwable {
    }

    static {
        FakeThing.registerNatives();
    }
}
````

**Comment by lospejos** [comment]
Thanks for comment, @leibnitz27 !
Unfortunately, I have a plenty of such naming collisions... Anyway I'll try to adopt this.
